restful.go

```diff
@@ -433,3 +461,69 @@ func cancelQuery(ctx context.Context, sr *snowflakeRestful, requestID string, ti
 		MessageArgs: []interface{}{resp.StatusCode, fullURL},
 	}
 }
+
+func getAsync(
+	ctx context.Context,
+	sr *snowflakeRestful,
+	headers map[string]string,
+	URL *url.URL,
+	timeout time.Duration,
+	res *snowflakeResult,
+	rows *snowflakeRows,
+) {
+	headers[headerAuthorizationKey] = fmt.Sprintf(headerSnowflakeToken, sr.Token)
+	resp, err := sr.FuncGet(ctx, sr, URL, headers, timeout)
+	if err != nil {
+		logger.WithContext(ctx).Errorf("failed to get response. err: %v", err)
+		return
+	}
+
+	respd := execResponse{} // reset the response
+	err = json.NewDecoder(resp.Body).Decode(&respd)
+	resp.Body.Close()
+	if err != nil {
+		logger.WithContext(ctx).Errorf("failed to decode JSON. err: %v", err)
+		return
+	}
```

Comment:
What's actually going to happen when this error is handled? Will rows.Next() block indefinitely?

You may wanna update statusChannel to not just receive a `queryStatus string` but a struct like:

```
type queryEvent struct {
  status queryStatus
  err error
}
```

Then update all of the checkAsyncExecStatus() functions to this signature:

```
checkAsyncExecStatus() error
```

Then update places like Next() to do something like this:

```
func (rows *snowflakeRows) Next(dest []driver.Value) (err error) {
    if err := rows.checkAsyncQueryStatus(); err != nil {
        return err
    }
```

That's just one idea though. The error handling flow would need to be tested (manually and I suggest writing unit tests for it as well).

Reply:
This makes sense. I'll update it as such.